beacon/goclient/goclient.go

```diff
@@ -187,6 +200,29 @@ func New(logger *zap.Logger, opt beaconprotocol.Options, operatorID spectypes.Op
 		zap.String("version", client.nodeVersion),
 	)
 
+	// If it's a Prysm, check that the debug endpoints are enabled, otherwise
+	// the Validators method might fail when calling BeaconState.
+	// TODO: remove this once Prysm enables debug endpoints by default.
+	if client.nodeClient == NodePrysm {
+		// BeaconState is quite heavy to call, checking ForkChoice should be enough.
+		resp, err := client.client.(eth2client.ForkChoiceProvider).ForkChoice(opt.Context, &api.ForkChoiceOpts{
+			Common: api.CommonOpts{Timeout: client.maxTimeout},
+		})
+		if err != nil {
+			var apiErr *api.Error
+			if errors.As(err, &apiErr) && apiErr.StatusCode == 404 {
```

Comment:
404 is well-known code, however perhaps this would be a bit more readable:

```suggestion
			if errors.As(err, &apiErr) && apiErr.StatusCode == http.StatusNotFound {
```

Reply:
done